### ISSUE TITLE
MacOS: Using official namespaces do not work - 418 "Problems" on default template

### ISSUE THREAD

**Issue by cushmatt** [OmniSharp/omnisharp-vscode#4140]
## Issue Description ##
This wasn't an issue until just recently.  Every dotnet core project is filled with omnisharp problems.  Noticed this on a project I opened up to continue working on for work (been about 2 weeks since I've worked on it and was working fine previously).  I decided to try creating a default project to see if the issue persisted there and it does.  Just creating a project with 
```bash
dotnet new webapp
``` 
shows 418 problems.  Odd thing is if I go to the project file and save it without any changes, the problems are reduced to 134.  Either way, these are a few of the example problems that are being reported:
```
The type or namespace name 'System' could not be found (are you missing a using directive or an assembly reference?) [dotnetcore3.1]
The type or namespace name 'Microsoft' could not be found (are you missing a using directive or an assembly reference?) [dotnetcore3.1]
The type or namespace name 'PageModel' could not be found (are you missing a using directive or an assembly reference?) [dotnetcore3.1]
```

In my project for work where I have 1k+ reported problems, the majority are like so:
```
The type 'Object' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [Eclypses.Senet.Server]
The type 'Object' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [Eclypses.Senet.Client]
The type 'IAsyncDisposable' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.1.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. [Eclypses.Senet.Client]
```

I have read up on this and tried multiple solutions to no avail.
1. Deleted and reinstalled the omnisharp extension located at `~/.vscode/extensions/ms-dotnettools.csharp-1.23.4`
2. Reinstalled dotnet core 3.1 from the Microsoft website
3. Added a reference to netstandard in my `.csproj` file(s) two different ways
    ```xml
    <!-- THIS METHOD -->
    <ItemGroup>
      <Reference Include="netstandard" />
    </ItemGroup>

    <!-- OR THIS MORE DETAILED METHOD -->
    <ItemGroup>
      <Reference Include="netstandard" Version="2.1.0.0" Culture="neutral" PublicKeyToken="cc7b13ffcd2ddd51" />
    </ItemGroup>

    <!-- OR THIS MORE DETAILED METHOD WITH VERSION 2.0 -->
    <ItemGroup>
      <Reference Include="netstandard" Version="2.0.0.0" Culture="neutral" PublicKeyToken="cc7b13ffcd2ddd51" />
    </ItemGroup>
    ```
4. I've tried modifying my `<TargetFramework>` tag to `<TargetFrameworks>` and adding in netstandard in there
    ```xml
    <!-- THIS -->
    <TargetFrameworks>netcoreapp3.1;netstandard2.1</TargetFrameworks>

    <!-- OR THIS -->
    <TargetFrameworks>netcoreapp3.1;netstandard2.0</TargetFrameworks>
    ```
    where I read and found out that Roslyn can't deal with multiple target frameworks and defaults to the first one listed
5. Ran `dotnet restore` which has no issues restoring but incurs the same problems listed in VSCode as well as red squiggly Intellisense errors/warnings all over every page that tries to use an something from an official namespace
6. Rebooted computer (hell, sometimes it works?)

The dotnet CLI incurs no issues whatsoever.  If I run `dotnet build`, it builds correctly with 0 errors/warnings.  I can `dotnet run` and run the application with no issues.  I can open in Visual Studio and incur no warnings or errors like I do in Visual Studio Code.  It's quite baffling to me.  I'm tempted to bootcamp over to Windows and see if the same issues occur.

## Steps to Reproduce ##
`dotnet new webapp`

## Expected Behavior ##
No problems reported by omnisharp

## Actual Behavior ##
Shows 418 problems and filling the screen with red IntelliSense squiggles everywhere that there is something using an official namespace. 

## Logs ##
When running `dotnet build`
```
Microsoft (R) Build Engine version 16.7.0+7fb82e5b2 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored /Users/cush/Documents/Development/Unofficial/dotnetcore3.1/dotnetcore3.1.csproj (in 101 ms).
  dotnetcore3.1 -> /Users/cush/Documents/Development/Unofficial/dotnetcore3.1/bin/Debug/netcoreapp3.1/dotnetcore3.1.dll
  dotnetcore3.1 -> /Users/cush/Documents/Development/Unofficial/dotnetcore3.1/bin/Debug/netcoreapp3.1/dotnetcore3.1.Views.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:03.75
```

### OmniSharp log ###
<details>
<summary>When first loaded up</summary>
Starting OmniSharp server at 10/22/2020, 4:46:35 PM
    Target: /Users/cush/Documents/Development/Unofficial/dotnetcore3.1

OmniSharp server started with Mono 6.12.0.
    Path: /Users/cush/.vscode/extensions/ms-dotnettools.csharp-1.23.4/.omnisharp/1.37.3/omnisharp/OmniSharp.exe
    PID: 5148

[info]: OmniSharp.Stdio.Host
        Starting OmniSharp on MacOS 10.15.6 (x64)
[info]: OmniSharp.Services.DotNetCliService
        DotNetPath set to dotnet
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Located 2 MSBuild instance(s)
            1: Mono 16.8.0 - "/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin"
            2: StandAlone 16.8.0 - "/Users/cush/.vscode/extensions/ms-dotnettools.csharp-1.23.4/.omnisharp/1.37.3/omnisharp/.msbuild/Current/Bin"
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Registered MSBuild instance: Mono 16.8.0 - "/Library/Frameworks/Mono.framework/Versions/6.12.0/lib/mono/msbuild/Current/bin"
            CscToolExe = csc.exe
            CscToolPath = /Users/cush/.vscode/extensions/ms-dotnettools.csharp-1.23.4/.omnisharp/1.37.3/omnisharp/.msbuild/Current/Bin/Roslyn
[info]: OmniSharp.Cake.CakeProjectSystem
        Detecting Cake files in '/Users/cush/Documents/Development/Unofficial/dotnetcore3.1'.
[info]: OmniSharp.Cake.CakeProjectSystem
        Could not find any Cake files
[info]: OmniSharp.MSBuild.ProjectSystem
        No solution files found in '/Users/cush/Documents/Development/Unofficial/dotnetcore3.1'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for '/Users/cush/Documents/Development/Unofficial/dotnetcore3.1/dotnetcore3.1.csproj'
[info]: OmniSharp.Script.ScriptProjectSystem
        Detecting CSX files in '/Users/cush/Documents/Development/Unofficial/dotnetcore3.1'.
[info]: OmniSharp.Script.ScriptProjectSystem
        Could not find any CSX files
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.CSharpFormattingWorkspaceOptionsProvider, Order: 0
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: /Users/cush/Documents/Development/Unofficial/dotnetcore3.1/dotnetcore3.1.csproj
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.Completion.CompletionOptionsProvider, Order: 0
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.RenameWorkspaceOptionsProvider, Order: 100
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.ImplementTypeWorkspaceOptionsProvider, Order: 110
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.BlockStructureWorkspaceOptionsProvider, Order: 140
[info]: OmniSharp.WorkspaceInitializer
        Configuration finished.
[info]: OmniSharp.Stdio.Host
        Omnisharp server running using Stdio at location '/Users/cush/Documents/Development/Unofficial/dotnetcore3.1' on host 4209.
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file '/Users/cush/Documents/Development/Unofficial/dotnetcore3.1/dotnetcore3.1.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Adding project '/Users/cush/Documents/Development/Unofficial/dotnetcore3.1/dotnetcore3.1.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: dotnetcore3.1
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for '/Users/cush/Documents/Development/Unofficial/dotnetcore3.1/dotnetcore3.1.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: /Users/cush/Documents/Development/Unofficial/dotnetcore3.1/dotnetcore3.1.csproj
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file '/Users/cush/Documents/Development/Unofficial/dotnetcore3.1/dotnetcore3.1.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: dotnetcore3.1
</details>

<details>
<summary>When deleting obj/ and bin/ directories and reloading the window</summary>
  Determining projects to restore...
  Restored /Users/cush/Documents/Work/repos/Eclypses.Senet/Eclypses.Senet.MTE/Eclypses.Senet.MTE.csproj (in 120 ms).
Done: 0.
  Determining projects to restore...
  Restored /Users/cush/Documents/Work/repos/Eclypses.Senet/Eclypses.Senet.Server/Eclypses.Senet.Server.csproj (in 621 ms).
  1 of 2 projects are up-to-date for restore.
Done: 0.
  Determining projects to restore...
  Restored /Users/cush/Documents/Work/repos/Eclypses.Senet/Eclypses.Senet.Client/Eclypses.Senet.Client.csproj (in 481 ms).
  1 of 2 projects are up-to-date for restore.
Done: 0.
</details>

### C# log ###
Nothing of note

## Environment information ##

**VSCode version**: 1.50.1
**C# Extension**: 1.23.4

<details><summary>Mono Information</summary>
        OmniSharp using global mono :6.12.0</details>
<details><summary>Dotnet Information</summary>
.NET Core SDK (reflecting any global.json):
 Version:   3.1.403
 Commit:    9e895200cd

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.15
 OS Platform: Darwin
 RID:         osx.10.15-x64
 Base Path:   /usr/local/share/dotnet/sdk/3.1.403/

Host (useful for support):
  Version: 3.1.9
  Commit:  774fc3d6a9

.NET Core SDKs installed:
  3.1.401 [/usr/local/share/dotnet/sdk]
  3.1.402 [/usr/local/share/dotnet/sdk]
  3.1.403 [/usr/local/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.1.7 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.8 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.9 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.21 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.22 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.7 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.8 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.9 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|
|---|---|---|
|august-themes|inci-august|2.1.0|
|auto-close-tag|formulahendry|0.5.9|
|auto-rename-tag|formulahendry|0.1.5|
|auto-using|Fudge|0.7.15|
|ayu|teabyii|0.20.1|
|azure-account|ms-vscode|0.9.3|
|azure-pipelines|ms-azure-devops|1.177.0|
|better-comments|aaron-bond|2.1.0|
|bracket-pair-colorizer-2|CoenraadS|0.2.0|
|code-settings-sync|Shan|3.4.3|
|code-spell-checker|streetsidesoftware|1.9.2|
|color-highlight|naumovs|2.3.0|
|comment|karyfoundation|7.0.3|
|csharp|ms-dotnettools|1.23.4|
|csharpextensions|jchannon|1.3.1|
|csharpfixformat|Leopotam|0.0.84|
|docomment|k--kato|0.1.18|
|eppz-code|eppz|1.2.52|
|es7-react-js-snippets|dsznajder|3.0.0|
|feather|dericcain|0.5.0|
|feppz--theme|foxnne|0.5.0|
|git-project-manager|felipecaputo|1.7.1|
|github-sharp-theme|joaomoreno|1.4.0|
|gitignore|codezombiech|0.6.0|
|gitlens|eamodio|10.2.2|
|js-quick-console|AhadCove|0.0.3|
|LiveServer|ritwickdey|5.6.1|
|markdown-pdf|yzane|1.4.4|
|markdown-preview-enhanced|shd101wyy|0.5.13|
|material-icon-theme|PKief|4.3.0|
|mayukaithemevsc|GulajavaMinistudio|1.5.2|
|mdx|silvenon|0.1.0|
|npm-intellisense|christian-kohler|1.3.1|
|one-dark-theme|mskelton|1.7.2|
|php-namespace-resolver|MehediDracula|1.1.8|
|polacode|pnp|0.3.4|
|pop-n-lock-theme-vscode|Luxcium|3.31.0|
|prettier-vscode|esbenp|5.7.1|
|project-manager|alefragnani|11.3.1|
|rainbow-csv|mechatroner|1.7.1|
|react-native-styled-component-snippets|milobo|1.0.1|
|reui|barrsan|1.1.4|
|rose-pine|mvllow|0.3.19|
|sort-lines|Tyriar|1.9.0|
|spaces-inside-braces|AiryShift|0.4.0|
|TabOut|albert|0.2.0|
|teppz|ofrades|2.4.2|
|theme-cobalt2|wesbos|2.1.6|
|theme-karyfoundation-themes|karyfoundation|20.0.3|
|theme-levelup|leveluptutorials|0.1.2|
|theme-monokai-pro-vscode|monokai|1.1.17|
|vim|vscodevim|1.17.1|
|vsc-community-material-theme|Equinusocio|1.4.2|
|vsc-material-theme|Equinusocio|33.0.0|
|vsc-material-theme-icons|equinusocio|1.2.0|
|vscode-edit-csv|janisdd|0.4.2|
|vscode-eslint|dbaeumer|2.1.13|
|vscode-faker|deerawan|1.4.1|
|vscode-graphql|GraphQL|0.3.10|
|vscode-great-icons|emmanuelbeziat|2.1.57|
|vscode-icons|vscode-icons-team|11.0.0|
|vscode-icontheme-nomo-dark|be5invis|1.3.6|
|vscode-import-cost|wix|2.12.0|
|vscode-intelephense-client|bmewburn|1.5.4|
|vscode-language-babel|mgmcdermott|0.0.30|
|vscode-markdownlint|DavidAnson|0.37.1|
|vscode-mdx-preview|xyc|0.3.0|
|vscode-npm-script|eg2|0.3.13|
|vscode-nuget-package-manager|jmrog|1.1.6|
|vscode-peacock|johnpapa|3.8.0|
|vscode-styled-components|jpoissonnier|1.2.0|
|vscode-xml|redhat|0.13.0|
|vscodeintellicode|VisualStudioExptTeam|1.2.10|
|vsliveshare|ms-vsliveshare|1.0.3046|;

</details>

## Conclusion ##
Apologize if there's something I'm not doing correctly in my report.  I rarely every make posts on GitHub issues as I usually always find the solution, but I've been working on this for days now and I'm just hoping this is solvable at this point.  I've been using Visual Studio to get my work done but I've got a 500 line VSCode settings.json file for a reason haha.

**Comment by JoeRobich** [comment]
@cushmatt Sorry about this issue. I had reverted the meaning of "useGlobalMono" to "always" because I was under the belief it shipped with MSBuild 16.8 but in fact it still ships with MSBuild 16.6. This is important because the 3.1.4xx SDKs require MSBuild 16.7 or higher. To fix you problem, set the "omnisharp.useGlobalMono" settings to "never" in VS Code and restart OmniSharp.

**Comment by cushmatt** [comment]
> @cushmatt Sorry about this issue. I had reverted the meaning of "useGlobalMono" to "always" because I was under the belief it shipped with MSBuild 16.8 but in fact it still ships with MSBuild 16.6. This is important because the 3.1.4xx SDKs require MSBuild 16.7 or higher. To fix you problem, set the "omnisharp.useGlobalMono" settings to "never" in VS Code and restart OmniSharp.
> 
> Edit: This setting will be default back to "never" in our next release.

Thank You!!!  That fixed it.  Sounds good, I'm happy it was a simple fix and not something more complex that I had to do in every `.csproj` file.
